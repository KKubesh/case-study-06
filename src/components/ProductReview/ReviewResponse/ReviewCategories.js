import React, {Component} from 'react';

class ReviewCatagories extends Component {

    render () {
        return (
            <div className="row-flex" data-testid="categories">
                <div className="categories-review" data-testid="categories-review">
                    <h3 data-testid="categories-pro">PRO</h3>
                    <h4 data-testid="categories-pro-text">most helpful 4-5 star review</h4>
                </div>
                <div className="categories-review" data-testid="categories-review">
                    <h3 data-testid="categories-con">CON</h3>
                    <h4 data-testid="categories-con-text">most helpful 1-2 star review</h4>
                </div>
            </div>   
        )
    }
}

export default ReviewCatagories;