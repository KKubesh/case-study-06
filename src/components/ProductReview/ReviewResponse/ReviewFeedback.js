import React, {Component} from 'react';
import redStar from '../../../images/red-star.png';
import greyStar from '../../../images/grey-star.png';

class ReviewFeedback extends Component {

    render () {
        return (
            <div className="row-flex" data-testid="review-feedback">
                <div className="review" data-testid="review">
                    <div className="reviewer" data-testid="reviewer">
                        <img src={redStar} alt="postive star 1"/><img src={redStar} alt="postive star 2"/><img src={redStar} alt="postive star 3"/><img src={redStar} alt="postive star 4"/><img src={redStar} alt="postive star 5"/>
                    </div>
                    <h3>Fantastic Blender</h3>
                    This blender works amazingly, and blends within seconds.  The single serve cups also work really well for smoothies or protein shakes!
                    <br/>
                    <br/>
                    <a href="https://www.google.com/search?q=eric">Eric</a> April 18, 2013
                </div>
                <div className="review" data-testid="review">
                    <div className="reviewer" data-testid="reviewer">
                        <img src={redStar} alt="postive star 1"/><img src={greyStar} alt="negative star 1"/><img src={greyStar} alt="negative star 2"/><img src={greyStar} alt="negative star 3"/><img src={greyStar} alt="negative star 4"/>
                    </div>
                    <h3>Very unhappy</h3>
                    Less than 2 months after purchase it completely stopped working. First it wouldn't detect the pitcher when trying to blend a significant amount, a couple weeks later it wouldn't detect the single serve cup.
                    <br/>
                    <br/>
                    <a href="https://www.google.com/search?q=New+York">New York</a> March 11, 2013
                </div>
            </div>  
        )
    }
}

export default ReviewFeedback;