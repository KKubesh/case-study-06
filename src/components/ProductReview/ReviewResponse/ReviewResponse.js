import React, {Component} from 'react';
import ReviewCatagories from './ReviewCategories';
import ReviewFeedback from './ReviewFeedback';

class ReviewResponse extends Component {

    render () {
        return (
            <div className="reviews" data-testid="review-response">
                <ReviewCatagories/>   
                <hr width="92%"/>
                <ReviewFeedback/> 
            </div>
        )
    }
}

export default ReviewResponse;