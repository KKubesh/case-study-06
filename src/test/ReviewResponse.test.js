import React from 'react';
import { render, cleanup } from 'react-testing-library';
import ReviewResponse from '../components/ProductReview/ReviewResponse/ReviewResponse';

afterEach(cleanup);

test('<ReviewResponse />', () => {
    const { getByTestId, container } = render(<ReviewResponse/>);
    expect(getByTestId('review-response')).toBeTruthy();
    expect(getByTestId('categories')).toBeTruthy();
    expect(getByTestId('review-feedback')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
});