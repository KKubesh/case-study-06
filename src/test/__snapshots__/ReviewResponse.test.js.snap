// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReviewResponse /> 1`] = `
<div
  class="reviews"
  data-testid="review-response"
>
  <div
    class="row-flex"
    data-testid="categories"
  >
    <div
      class="categories-review"
      data-testid="categories-review"
    >
      <h3
        data-testid="categories-pro"
      >
        PRO
      </h3>
      <h4
        data-testid="categories-pro-text"
      >
        most helpful 4-5 star review
      </h4>
    </div>
    <div
      class="categories-review"
      data-testid="categories-review"
    >
      <h3
        data-testid="categories-con"
      >
        CON
      </h3>
      <h4
        data-testid="categories-con-text"
      >
        most helpful 1-2 star review
      </h4>
    </div>
  </div>
  <hr
    width="92%"
  />
  <div
    class="row-flex"
    data-testid="review-feedback"
  >
    <div
      class="review"
      data-testid="review"
    >
      <div
        class="reviewer"
        data-testid="reviewer"
      >
        <img
          alt="postive star 1"
          src="red-star.png"
        />
        <img
          alt="postive star 2"
          src="red-star.png"
        />
        <img
          alt="postive star 3"
          src="red-star.png"
        />
        <img
          alt="postive star 4"
          src="red-star.png"
        />
        <img
          alt="postive star 5"
          src="red-star.png"
        />
      </div>
      <h3>
        Fantastic Blender
      </h3>
      This blender works amazingly, and blends within seconds.  The single serve cups also work really well for smoothies or protein shakes!
      <br />
      <br />
      <a
        href="https://www.google.com/search?q=eric"
      >
        Eric
      </a>
       April 18, 2013
    </div>
    <div
      class="review"
      data-testid="review"
    >
      <div
        class="reviewer"
        data-testid="reviewer"
      >
        <img
          alt="postive star 1"
          src="red-star.png"
        />
        <img
          alt="negative star 1"
          src="grey-star.png"
        />
        <img
          alt="negative star 2"
          src="grey-star.png"
        />
        <img
          alt="negative star 3"
          src="grey-star.png"
        />
        <img
          alt="negative star 4"
          src="grey-star.png"
        />
      </div>
      <h3>
        Very unhappy
      </h3>
      Less than 2 months after purchase it completely stopped working. First it wouldn't detect the pitcher when trying to blend a significant amount, a couple weeks later it wouldn't detect the single serve cup.
      <br />
      <br />
      <a
        href="https://www.google.com/search?q=New+York"
      >
        New York
      </a>
       March 11, 2013
    </div>
  </div>
</div>
`;
